lib/awful/prompt.lua

```diff
@@ -518,10 +519,10 @@ function prompt.run(args, textbox, exe_callback, completion_callback,
                 hooks[key] = hooks[key] or {}
                 hooks[key][#hooks[key]+1] = v
             else
-                assert("The hook's 3rd parameter has to be a function.")
+                debug.print_warning("The hook's 3rd parameter has to be a function.")
             end
         else
-            assert("The hook has to have 3 parameters.")
+            debug.print_warning("The hook has to have 3 parameters.")
         end
```

Comment:
My opinion is still that code that didn't cause an error in 4.0 should not start to cause errors in 4.1. Beside, as I said on IRC, a warning is enough. It won't break anything if the table element count is wrong. Having an error is over-reacting.